src/Input/Silk.NET.Input.Glfw/GlfwInputContext.cs

```diff
@@ -50,30 +50,48 @@ public unsafe GlfwInputContext(IView window)
             Mice = _mice;
 
             GlfwInputPlatform.RegisterWindow((WindowHandle*) Handle, _subscribers);
-            window.Update += _update = _ =>
+            if (window is ViewImplementationBase view)
             {
-                foreach (var updatable in _mice)
-                {
-                    updatable.Update();
-                }
-
-                foreach (var updatable in _gamepads)
-                {
-                    updatable.Update();
-                }
-
-                foreach (var updatable in _joysticks)
-                {
-                    updatable.Update();
-                }
-            };
+                view.ProcessEvents += ProcessEvents;
+            }
+            else
+            {
+                window.Update += Update;
+            }
 
             _window = window;
         }
 
+        private void Update(double delta) => ProcessEvents();
+        private void ProcessEvents()
+        {
+            foreach (var updatable in _mice)
+            {
+                updatable.Update();
+            }
+
+            foreach (var updatable in _gamepads)
+            {
+                updatable.Update();
+            }
+
+            foreach (var updatable in _joysticks)
+            {
+                updatable.Update();
+            }
+        }
+
         public unsafe void Dispose()
         {
-            _window.Update -= _update;
+            if (_window is ViewImplementationBase view)
+            {
+                view.ProcessEvents += ProcessEvents;
+            }
+            else
+            {
+                _window.Update += Update;
```

Comment:
Shouldn't these be removing the event handlers rather than adding them again?

Reply:
Yes, thanks!

```suggestion
                _window.Update -= Update;
```